libkbfs/folder_branch_ops.go

```diff
@@ -1206,6 +1206,15 @@ func (fbo *folderBranchOps) initMDLocked(
 		return err
 	}
 
+	fbo.headLock.Lock(lState)
+	defer fbo.headLock.Unlock(lState)
+	// Some other thread got here first, so give up and let it go
+	// before we push anything to the servers.
```

Comment:
Yeah I guess that works.  I didn't think it was a big deal to hold it over network access when the folder still hadn't been initialized yet, since any possible other call will be blocking on that initialization anyway (because either they block on `mdWriterLock`, or they block on `getRootNode()` which ends up blocking on `mdWriterLock` in that case. It felt safer to me, but now I remember that `setHeadLocked` asserts on both locks, so I'll change it.

But I can